lib/patch-series.ts

```diff
@@ -718,9 +740,42 @@ export class PatchSeries {
             if (footers.length > 0) {
                 footers.push(""); // empty line
             }
-            // split the range-diff and prefix with a space
-            footers.push(`Range-diff vs v${this.metadata.iteration - 1}:\n\n${
-                         this.rangeDiff.replace(/(^|\n(?!$))/g, "$1 ")}\n`);
+
+            let rangeDiff = "";
+
+            if (typeof this.options.versionDiff === "undefined" || this.options.versionDiff) {
+                const args = [ this.rangeDiff.previousRange ];
+
+                if (this.rangeDiff.currentRange) {
+                    args.push(this.rangeDiff.currentRange);
+                }
+
+                // please set rangeDiff empty if post analysis decides it is too big or messy
+                rangeDiff = await git(["range-diff", "--no-color", "--creation-factor=95", ...args],
+                    { workDir: this.project.workDir });
+            }
+
+            if (rangeDiff) {
+                // split the range-diff and prefix with a space
+                footers.push(`Range-diff vs v${this.metadata.iteration - 1}:\n\n${
+                            rangeDiff.replace(/(^|\n(?!$))/g, "$1 ")}\n`);
+            } else {
+                // rev-parse does not have info for shallow repos
+                const getRange = (range: string): string => {
+                    const hashes = range.match(/([a-z,0-9]{7}).+?(\.+)([a-z,0-9]{7})/);
+                    if (hashes) {
+                        return `${hashes[1]}${hashes[2]}${hashes[3]}`;
+                    } else {
+                        throw Error(`Range parse failed for ${range}`);
+                    }
+                };
+
+                footers.push(`Range-diff too large. Please review it locally by calling:
+   git fetch https://github.com/gitgitgadget/git ${this.rangeDiff.baseCommit.substring(0, 7)} ${
+                    this.rangeDiff.headCommit.substring(0, 7)}
```

Comment:
I would like to avoid hard-coding the `7` here, maybe we can do something like adding `gitShortHash(revision)` to `lib/git.ts` and use that here? Oh, and increase the length to 8, to stay on the safe side.

Reply:
Please see the updates.